### PR TITLE
SRCH-3318 upgrade Tika to 2.4.1

### PR DESCRIPTION
**Pull request by MothOnMars**
## Summary
This PR:
- updates our document metadata parsing for compatibility with Tika 2.4.1
- adds & updates unit tests
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks

- [x] You have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/models/application_document.rb
Comment (lorawoodford) [inline app/models/application_document.rb:27]: It's a preference thing so I'm not going to block on this, but I find:

```
    content = metadata['X-TIKA:content']
    return '' unless content&.match?(/([a-zA-Z])|\d/)

    content.tr("\uFFFD", ' ')&.squish
```

to be slightly more readable since it's rare to see that mutable string construction in the wild and since you've added the return there that can catch the nil content if one exists.  But again, totally a preference call.

**Review by lorawoodford — APPROVED** [review]
One comment left in-line, but also approving as-is if you disagree.  LGTM.

File: app/models/application_document.rb
Comment (MothOnMars) [reply app/models/application_document.rb]: good call - will do

**Review by lorawoodford — DISMISSED** [review]
LGTM

**Comment by MothOnMars** [comment]
@lorawoodford , this needs a re-review. I had to tweak the specs after merging `main` to make them compatible with Tika 2.4.

**Review by lorawoodford — APPROVED** [review]
Sorry for the delay.  LGTM.

**Pull request merged**
